modules/ghlint-api/src/main/kotlin/net/twisterrob/ghlint/model/Permission.kt

```diff
@@ -0,0 +1,37 @@
+package net.twisterrob.ghlint.model
+
+public enum class Permission(public val value: String) {
+	ACTIONS("actions"),
+	ATTESTATIONS("attestations"),
+	CHECKS("checks"),
+	CONTENTS("contents"),
+	DEPLOYMENTS("deployments"),
+	ID_TOKEN("id-token"),
+	ISSUES("issues"),
+	METADATA("metadata"),
+	PACKAGES("packages"),
+	PAGES("pages"),
+	PULL_REQUESTS("pull-requests"),
+	REPOSITORY_PROJECTS("repository-projects"),
+	SECURITY_EVENTS("security-events"),
+	STATUSES("statuses")
+}
+
+public enum class Access(public val value: String) : Comparable<Access> {
+	// Order determines restrictiveness of permission
+	NONE("none"),
+	READ("read"),
+	WRITE("write");
+
+	public companion object {
+		public fun fromString(value: String): Access = when (value) {
+			"read" -> READ
+			"write" -> WRITE
+			else -> NONE
+		}
+	}
+}
+
+public data class Scope(val permission: Permission, val access: Access) {
```

Comment:
remove data and generate hashCode and equals

Reply:
```suggestion
public class Scope(
    public val permission: Permission,
    public val access: Access,
) {
```

Reply:
or make internal? is it only used in the rule?

Reply:
It's used in the extensions module + in the rule module.

Is there a "good" way to generate the hashCode and equals? Or do I just write the implementations?

I used `data class` so that I _didn't_ need to do this

Reply:
IDEA > menu > Code > Generate... is good enough. `@Poko` would be an alternative, but don't want to add it just yet. Also Kotlin 2.x (probably in a year or two will remove the componentX funs from data classes and that will make Poko obsolete!)